src/main/java/uk/gov/hmcts/reform/hmc/api/services/NextHearingDetailsServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+public class NextHearingDetailsServiceImpl implements NextHearingDetailsService {
+
+    private static Logger log = LoggerFactory.getLogger(HearingsServiceImpl.class);
+
+    @Autowired PrlUpdateService prlUpdateService;
+
+    @Override
+    public Boolean updateNextHearingDate(Hearings hearings) {
+        log.info("inside  updateNextHearingDateInCcd .... ");
+        NextHearingDetails nextHearingDetails = getNextHearingDate(hearings);
+        Boolean isNextHearingDateUpdated = false;
+        if (null != nextHearingDetails) {
+            log.info(
+                    "Next Hearing Date Details - ID {} and Date {} ",
+                    nextHearingDetails.getHearingId(),
+                    nextHearingDetails.getNextHearingDate());
+            NextHearingDetailsDTO nextHearingDateDetailsDTO =
+                    NextHearingDetailsDTO.nextHearingDetailsRequestDTOWith()
+                            .nextHearingDetails(nextHearingDetails)
+                            .caseRef(hearings.getCaseRef())
+                            .build();
+            isNextHearingDateUpdated =
+                    prlUpdateService.updatePrlServiceWithNextHearingDate(nextHearingDateDetailsDTO);
+        }
+        return isNextHearingDateUpdated;
+    }
+
+    @Override
+    public State fetchStateForUpdate(Hearings hearings, String currHearingHmcStatus) {
+        log.info("test state .... {}", DECISION_OUTCOME);
+        Boolean isAllCompleted =
+                hearings.getCaseHearings().stream()
+                                .filter(
+                                        eachHearing ->
+                                                eachHearing.getHmcStatus().equals(COMPLETED)
+                                                        || eachHearing
+                                                                .getHmcStatus()
+                                                                .equals(ADJOURNED))
+                                .collect(Collectors.toList())
+                                .size()
+                        == hearings.getCaseHearings().size();
+
+        if (isAllCompleted) {
+            return DECISION_OUTCOME;
+        } else {
+            if (currHearingHmcStatus.equals(COMPLETED) || currHearingHmcStatus.equals(ADJOURNED)) {
+                return anyFutureHearings(hearings)
+                        ? PREPARE_FOR_HEARING_CONDUCT_HEARING
+                        : DECISION_OUTCOME;
+            } else {
+                return PREPARE_FOR_HEARING_CONDUCT_HEARING;
+            }
+        }
+    }
+
+    private Boolean anyFutureHearings(Hearings hearings) {
+        for (CaseHearing hearing : hearings.getCaseHearings()) {
+            Optional<LocalDateTime> minDateOfHearingDaySche =
+                    hearing.getHearingDaySchedule().stream()
+                            .filter(u -> u.getHearingStartDateTime().isAfter(LocalDateTime.now()))
+                            .map(u -> u.getHearingStartDateTime())
```

Comment:
is this required?

Reply:
It is required in order to calculate the latest or near future hearing, out of so many hearingDaySchedule.

Reply:
Not required, I have optimised it. Resolved